buildrunner/docker/multiplatform_image_builder.py

```diff
@@ -212,6 +216,46 @@ def _stop_local_registry(self):
         else:
             logger.warning("Local registry is not running when attempting to stop it")
 
+    # pylint: disable=too-many-arguments,too-many-locals
+    def _build_with_inject(self,
+                           inject: dict,
+                           tagged_names: List[str],
+                           platform: str,
+                           push: bool,
+                           path: str,
+                           file: str,
+                           build_args: dict,) -> None:
+
+        if not path or not os.path.isdir(path):
+            logger.warning(f"Failed to inject {inject} for {tagged_names} since path {path} isn't a directory.")
+            return
+
+        dir_prefix = 'mp-tmp-dir'
+        with tempfile.TemporaryDirectory(dir=self._temp_dir, prefix=dir_prefix) as tmp_dir:
+
+            context_dir = os.path.join(tmp_dir, f'{dir_prefix}/')
+            shutil.copytree(path, context_dir, ignore=shutil.ignore_patterns(dir_prefix, '.git'))
```

Comment:
Is this how the other code works? I thought there was a .dockerignore file that was handled as well, but maybe docker just does that on its own?

Reply:
There is no `.dockerignore` file in this repo. 

Single platform builds puts the build context in a tarfile and passes it to docker-py in the fileobj arg which may be like the build-context arg for `docker build`, it's a little hazy. Python on whales doesn't support fileobj or --build-context, it needs to be in the `path`. Copying the entire path dir and injecting the added files into the tmp dir was the way I was able to get it to work. I am open to more efficient ways since I don't love that it is copying everything.